### PR TITLE
Add support for recent PostgreSQL versions in Metricbeat

### LINKED ISSUE #23672: [docs] Obsolete postgresql database metricset generated document example
[PostgresSQL database metricset](https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-metricset-postgresql-database.html#_fields_173) shows an obsolete generated document example.

For example:
```
    "beat": {
        "hostname": "host.example.com",
        "name": "host.example.com"
    },
    "metricset": {
        "host": "postgresql:5432",
        "module": "postgresql",
        "name": "database",
        "rtt": 115
    },
```
`beat` has been removed, and `metricset` has changed to `event` with different fields.
& other changes...

For confirmed bugs, please report:
- Version: 7.10
- Operating System: 
- Discuss Forum URL:
- Steps to Reproduce:

### PR DESCRIPTION
**Pull request by jsoriano**
## What does this PR do?

* Run system tests with PostgreSQL 10, 11, 12 and 13.
* Run integration tests with PostgreSQL 13.
* Add fields available since PostgreSQL 10 in the activity metricset to give more details about the backend and its waits. Add example event for backend entries.
* Since PostgreSQL 12, `pg_stat_database` includes an entry for shared resources that don't belong to a single database. Handle this in tests and add docs and add example document.
* Handle metrics renamed in `pg_stat_statements` in PostgreSQL 13.

## Why is it important?

Improve support of recent versions of PostgreSQL.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] ~~I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Fix #23672
- Part of https://github.com/elastic/beats/issues/14337
- Part of https://github.com/elastic/integrations/issues/366

### CODE REVIEW

File: metricbeat/module/postgresql/activity/activity.go [hunk 1]
```diff
@@ -56,13 +57,19 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
...
 	ctx := context.Background()
+
 	results, err := m.QueryStats(ctx, "SELECT * FROM pg_stat_activity")
 	if err != nil {
 		return errors.Wrap(err, "error in QueryStats")
 	}
 
 	for _, result := range results {
-		data, _ := schema.Apply(result)
+		var data common.MapStr
```
[the remark this answers is not in the document]
Comment (jsoriano) [reply metricbeat/module/postgresql/activity/activity.go:67]: Good point, added.

File: metricbeat/module/postgresql/statement/statement.go [hunk 2]
```diff
@@ -66,6 +66,10 @@ func (m *MetricSet) Fetch(reporter mb.ReporterV2) error {
 
 	for _, result := range results {
 		data, _ := schema.Apply(result)
+		if _, ok := result["total_time"]; ok {
```
Comment (sayden) [inline metricbeat/module/postgresql/statement/statement.go:69]: Maybe I'll also add a comment here, for a contributor or someone new to the code, to help navigate at first sight.

File: metricbeat/mb/testing/fetcher.go [hunk 3]
```diff
@@ -96,6 +96,8 @@ func (f *reportingMetricSetV2FetcherError) WriteEvents(t testing.TB, path string
 }
 
 func (f *reportingMetricSetV2FetcherError) WriteEventsCond(t testing.TB, path string, cond func(common.MapStr) bool) {
+	t.Helper()
```
Comment (sayden) [inline metricbeat/mb/testing/fetcher.go:99]: Just a heads up in case this line slipped here.

**Review by sayden — APPROVED** [review]
Overall LGTM. Just left few non-blocking comments to help comprehension of some parts. Thanks for this effort!

File: metricbeat/module/postgresql/statement/data.go [hunk 4]
```diff
@@ -62,3 +62,17 @@ var schema = s.Schema{
 		},
 	},
 }
+
+// PostgreSQL 13 renames fields with exec times, keep this for compatibility with older versions.
```
[the remark this answers is not in the document]
Comment (jsoriano) [reply metricbeat/module/postgresql/statement/data.go:66]: I have rephrased this to be more specific about the renamed fields.

File: metricbeat/mb/testing/fetcher.go (hunk 3 above)
Comment (jsoriano) [reply metricbeat/mb/testing/fetcher.go:99]: This is intentionally added, so errors are more meaningful. This is not really related, so I can move this line to a different PR if you prefer.

File: metricbeat/module/postgresql/statement/statement.go (hunk 2 above)
Comment (jsoriano) [reply metricbeat/module/postgresql/statement/statement.go:71]: Added.

**Pull request merged**